lib/home_page.dart

```diff
@@ -28,8 +32,47 @@ class _HomePageState extends State<HomePage> {
 
   @override
   Widget build(BuildContext context) {
-    return Scaffold(
-      body: Container(),
+    return SafeArea(
+      child: Scaffold(
+        backgroundColor: Colors.white,
+        appBar: AppBar(
+          automaticallyImplyLeading: false,
+          backgroundColor: Colors.red.shade900,
+          title: Image.asset(
+            StringConstants.logoImage,
+            width: NumericConstants.logoImageWidth,
+          ),
+          centerTitle: true,
+        ),
+        body: StreamBuilder<List<Pokemon>>(
+            stream: _pokemonBloc.allPokemonsStream,
+            builder: (context, AsyncSnapshot<List<Pokemon>> snapshot) {
+              if (snapshot.hasData) {
+                return buildList(
+                  snapshot.data!,
+                );
+              } else if (snapshot.hasError) {
+                return Text(
+                  snapshot.error.toString(),
+                );
+              }
+              return Center(
+                child: CircularProgressIndicator(),
+              );
+            }),
+      ),
+    );
+  }
+
+  Widget buildList(List<Pokemon> data) {
+    return GridView.builder(
+      itemCount: data.length,
+      gridDelegate: SliverGridDelegateWithFixedCrossAxisCount(
+        crossAxisCount: NumericConstants.crossAxisCount,
+      ),
+      itemBuilder: (BuildContext context, int index) {
+        return GridTilePokemons(data: data, index: index);
```

Comment:
why do you pass as parameter all the list of pokemons and the index, instead to use only one element?
Like the widget is call GridTile i think you need to use only one pokemon like data[index]